### ISSUE TITLE
Fails to open file with cyrillic symbols in path

### ISSUE THREAD

**Issue by Andryshik345** [fstl-app/fstl#122]
fstl fails to open file if there are any cyrillic symbols in path, but it opens them just fine when using `file - open`, or drag-and-drop.
![k3nxmv](https://github.com/user-attachments/assets/de446965-7b63-4850-8b9c-c7b0cdfe967c)

OS: Windows 7 SP1, so maybe on windows 10 it works correctly?

**Comment by DeveloperPaul123** [comment]
Can you post an example file?

**Comment by Andryshik345** [comment]
> Can you post an example file?

I don't think it really matters, since I can pick two copies of the same file and rename it. You also can try to pick any of your STL files and name it something like `тестовая модель.stl`.
It also doesn't matter if it is `D:\фуу\bar.stl` or `D:\foo\бар.stl`>\.

I'll also check on Windows 10 OS later.

**Comment by Andryshik345** [comment]
Yep, fails on Windows 10 as well.

![изображение](https://github.com/user-attachments/assets/786b882b-e29d-4fa8-b5ad-3d3a3c1017cd)

**Comment by FredrikAV** [comment]
This is true also for Swedish but it seems it can easily be fixed by replacing the line (in app.cpp)

QString filename = argv[1];

with the lines

QStringList args = QCoreApplication::arguments();
QString filename = args.at(1); // Again assuming the file path is the first argument

Works for my binary at least.

**Comment by FredrikAV** [comment]
For anyone interested, there seems to be a system setting in Windows 10 that solves this too (i.e. without needing fix mentioned above):


If you're on Windows 10, check this:

Open intl.cpl (Run → intl.cpl)

Go to the Administrative tab

Click "Change system locale…"

In the dialog, is “Beta: Use Unicode UTF-8 for worldwide language support” checked?

If it’s not, check it and reboot. This can fix weird UTF-8 issues system-wide (especially for older apps or tools using Win32 APIs improperly).

WARNING: If there are any side effects to this I do not know, but from my tests at least it is reversable (i.e. you can turn it off again to get the same system behaviour as before).

**Comment by DeveloperPaul123** [comment]
I can't reproduce this on Win 11, but will patch with the suggested change in this issue.

**Comment by DeveloperPaul123** [comment]
Actually, I was testing the wrong thing. The use of `QCoreApplication::arguments()` does indeed fix things. See #129.
